src/Slices/ServiceInstanceDetails/Utils/AttributeUtils.ts

```diff
@@ -104,26 +107,50 @@ export const formatTreeRowData = (
         value: value,
         id: path + key,
         name: key,
+        type: "Value",
         children: [],
       };
 
+      // we need to check if the item is an inter-service relation
+      const relationNode = serviceModel.inter_service_relations?.find(
+        (relation) => relation.name === key
+      );
+
+      if (relationNode) {
+        node.type = "Relation";
+        node.serviceName = relationNode.entity_type;
+      }
+
       // If we are facing an object (Arrays are also considered as objects in JS),
       // it means we need to display a collapsible section.
       // The children property tells whether a section is a value, or a collapsible section.
-      if (value && typeof value === "object") {
+      if (value && typeof value === "object" && !relationNode) {
+        node.type = "Embedded";
+
         // In case we are dealing with an array of arrays, we want to keep track of the index to display it in the table.
         if (Array.isArray(value)) {
           for (let index = 0; index < value.length; index++) {
             const item = value[index];
+            let model: Partial<ServiceModel> = serviceModel;
 
+            // If the item is an object, we need to add the properties into the children
             if (typeof item === "object" && item !== null) {
-              // If the item is an object, we need to add the properties into the children
+              const embeddedNode = model.embedded_entities?.find(
```

Comment:
I think we can add model.embedded_entities to assertion and remove `?`

Reply:
No because there are not always embedded entities, but still can be a dict :D so we don't set the "Embedded" type